cmd/dockerd/daemon.go

```diff
@@ -277,6 +286,7 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 	errAPI := <-serveAPIWait
 	c.Cleanup()
 	shutdownDaemon(d)
+	close(watchStream)
```

Comment:
If it is not closed, the `ProcessClusterNotifications` goroutine will never exit.